ospin-web-dev/hambda
# FIX HISTORY

* adds helper models and redis client

* migrates the event trigger code from cortex to a new eventTrigger service

* Removes event triggers from cortex

* adds new service to circleCI, codecov and package-scripts

* deactivate loading processes from cache until cache invalidation is implemented

* changes subnet and fixes topic for event publishing

* changed message creation and assoaciated streamId

* adjust displayMsg to consider set name alias and chosen unit

* allow streamId on logs

* adjust Log service and test to allow streamId on logs

* default streamId to null in JOI and adds streamId to the Event schema

* code smell fixes after review

* bumps serverless-domain-manager minor version to resolve config issue

* publishes to correct topic to create event

* adds event display names to map and renames limit to threshold

* moves event triggers to top level key to allow all upstreams to have event triggers

Co-authored-by: Viktor Diezel <viktor.diezel@ospin.de>

fixes missing DB connection (#400)

removes dashboardStreams references in the code base (#405)

356 removes devices key on process (#406)

* adds remove field script for database migration

* changes the signature of add and remove field to make use if db connection

* adds tests

breaks up device test file and lints them (also renames command.test.js into openSsh.test.js) (#409)

breaks up user test file and lints them (#408)

Co-authored-by: Daniel Seehausen <Daniel.Seehausen@gmail.com>

379 abstract user model sausage hidden (#410)

* misc. stuff - I have only ever seen comma dangle warnings where we need to put it in

* Adds new ResourceAccess domain; abstracts some of 'access-rights-lib' into it; adds test coverage

* renames files to match tree/domain; user delete test got an overhaul, but I dont know why it is showing as a new file...maybe I added it I dont remember doing that though

* Adds user notifications interface

* adds helper method for container start up. Connects the DB and refers all models to said DB

* minor test updates and increases to coverage

* adds user model/interface; removes old user model/interface. The MEAT of this PR

* TODO no longer relevant with the new ResourceAccess interface, which provides helper methods to address this

* minor addition to validation lib

* warn loudly if a connection is being overwritten

* Stubs methods that will be a part of the Process and Device model PRs. These were ones needed for the User changes

* minor changes to test helpers to make use of Access

* updates device to use new Access

* updates process to use new Access

* minorly edits user JOI

* updates delete user to use a bundle of cross domain operations (called a  for now); updates tests for broader coverage

* adds the rest of the controller updates. these all make use of the User and other models, and take certain validations/assertions out of the old model.

* makes PR review requests; most notably edits access to only allow a user in one group at any given time - see test changes for documentation

* fixes broken tests

Staging (#402)

event trigger service and quick fix for missing UIConfig DB connection

Make tests async  (#403)

* Make tests async + merge with master

* Making dep install async

* test commit if it fails on circleci

* Remove failing test

383 fixes failing log client error on sequential invocations (#416)

* removes all deviceName from process model

* removes DoNotWaitForEmptryEventLoop from the middleware

Co-authored-by: Viktor Diezel <viktor.diezel@ospin.de>

only loads seeder if in test environment

bumps the node runtime (#421)

Co-authored-by: Viktor Diezel <viktor.diezel@ospin.de>

390 device admins can run all commands (#422)

* allows admin access to end process; adds Joi to it and tests

* adds admin access to the pause command; adds test and Joi

* adds test, joi and admin access to next phase and resume process; fixes names

* unifies tests for startProcess; adds Joi and admin access

* dries command joi schemas

Co-authored-by: Viktor Diezel <viktor.diezel@ospin.de>

fix joi schema to make entryPhaseId optional (#429)

Co-authored-by: Viktor Diezel <viktor.diezel@ospin.de>

Kill cortex (#423)

* migrates data storage functionality from cortex to new service called data

* kill cortex

* renames the service from data to dataPoints

Co-authored-by: Viktor Diezel <viktor.diezel@ospin.de>
Co-authored-by: Daniel Seehausen <Daniel.Seehausen@gmail.com>

allows strings now for stream defaults (support for oneOne) (#430)

Co-authored-by: Viktor Diezel <viktor.diezel@ospin.de>
Co-authored-by: O-Susanoo <57438770+O-Susanoo@users.noreply.github.com>

Jsdocs for events (#420)

* adds dependencies and scripts for jsdocs; updates gitignore

* removes unsued file

* implements the documentation for all mqtt communciation

* Adds CD integration

* cleans up <br> tags (not necessary) and removes JSON hint for mqtt messages

Co-authored-by: Viktor Diezel <viktor.diezel@ospin.de>
Co-authored-by: O-Susanoo <57438770+O-Susanoo@users.noreply.github.com>
